### ISSUE TITLE
Discuss diagnostic behavior for local resource usage within statically known unreachable code

### ISSUE THREAD

**Issue by bob80905** [llvm/wg-hlsl#427]
Consider the below shader:

```
RWByteAddressBuffer gBuf0 : register(u0);
RWByteAddressBuffer gBuf1 : register(u1);

uint Pass_EarlyReturn(bool cond, uint idx) {
    RWByteAddressBuffer buf = gBuf0;

    if (cond)
        buf.Store(idx * 4, 31);

        return 31;
    buf = gBuf1;
    buf.Store(idx * 4, 31);

    return 31;
}

[numthreads(8,8,1)]
void main(uint3 tid : SV_DispatchThreadID) {
    uint idx = tid.x + tid.y * 8;
    Pass_EarlyReturn(true, idx);
}
```
A diagnostic warning is currently emitted upon assigning gBuf1 to buf, even though this is dead code.
DXC does not emit a diagnostic, while Clang currently emits a warning diagnostic.
Clang could copy DXC and not emit a warning diagnostic for dead code, or it could be even stricter and emit an error in this case.
This issue tracks the HLSL team making a design decision on how to handle diagnostics for dead code, especially for diagnostics due to local resources.
